packages/device-connector/src/deployment-server/connection.ts

```diff
@@ -79,23 +82,36 @@ export const openStream = async (): Promise<Duplex> => {
   let connectorData: ConnectorData | undefined
   if (fs.existsSync('.connection.data')) {
     connectorData = await readConnectorData()
+    logger.debug(`Found previous connection on ${connectorData.uri} with id ${connectorData.id}`)
   }
 
-  if ((connectorData == null) || connectorData.uri !== address) {
-    connectorData = await generateConnectorData()
-    await writeConnectorData(connectorData)
+  if (connectorData == null || connectorData.id == null || connectorData.uri !== deployUri) {
+    logger.debug('No or invalid previous connection found for requested deployment-server')
+    try {
+      connectorData = await generateConnectorData()
+    } catch (e) {
+      logger.error(e)
+      await setTimeout(3000)
```

Comment:
Might want to configure this via ENV in the future

Reply:
Possible change for future changes imo